src/SVDiagnosticCollection.ts

```diff
@@ -182,51 +175,26 @@ export class SVDiagnosticCollection implements Disposable {
      * Also goes through the codeflow locations, to update the locations
      */
     public async mappingChanged(): Promise<void> {
-        for (const key of this.issuesCollection.keys()) {
-            const issues: SarifViewerVsCodeDiagnostic[] | undefined = this.issuesCollection.get(key);
-            if (!issues) {
-                continue;
+        for (const issues of this.issuesCollection.values()) {
+            for (const issue of issues) {
+                await SVDiagnosticFactory.tryToRemapLocations(this.fileMapper, issue);
             }
-
-            for (const index of issues.keys()) {
-                await SVDiagnosticFactory.tryToRemapLocations(this.explorerController, issues[index]);
-            }
-        }
-
-        const explorerDiag: SarifViewerVsCodeDiagnostic | undefined = this.explorerController.activeDiagnostic;
-        if (!explorerDiag) {
-            return;
         }
 
-        for (const key of this.unmappedIssuesCollection.keys()) {
-            const issues: SarifViewerVsCodeDiagnostic[] | undefined = this.unmappedIssuesCollection.get(key);
-            if (!issues) {
-                return;
-            }
-
+        for (const [key, unmappedIssues] of this.unmappedIssuesCollection.entries()) {
             const remainingUnmappedIssues: SarifViewerVsCodeDiagnostic[] = [];
-            for (const index of issues.keys()) {
-                const diag: SarifViewerVsCodeDiagnostic = issues[index];
-                await SVDiagnosticFactory.tryToRemapLocations(this.explorerController, diag).then((remapped) => {
-                    if (remapped) {
-                        this.add(diag);
-                        this.diagnosticCollectionChangedEventEmitter.fire({
-                            diagnostics: [diag],
-                            type: 'Add'
-                        });
-                        if (explorerDiag !== undefined && explorerDiag.resultInfo.runId === diag.resultInfo.runId &&
-                            explorerDiag.resultInfo.id === diag.resultInfo.id) {
-                            this.explorerController.setActiveDiagnostic(diag, true);
-                        }
```

Comment:
Couple of comments for this change. First, the "add" function already fires the event, so no need to fire it twice. Second, setting the active diagnostic in the explorer view in a loop is actually not good as if the remapping changes multiple diagnostics, the loop will continue to reset it.

We will need to think about whether a successful mapping should change the active diagnostic in the web-view in the first place.